runtime/src/message_processor.rs

```diff
@@ -316,10 +323,13 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
             return Err(InstructionError::ReentrancyNotAllowed);
         }
         self.program_ids.push(*key);
-        Ok(())
+        let mut keyed_accounts = unsafe { std::mem::transmute(keyed_accounts) };
```

Comment:
I'd really like to see all the unsafe code gone, we've spent a lot of care to avoid unsafe code in the runtime (besides the program's ffi boundaries).  There has to be a better way.  Maybe `Rc<RefCell>` or some other Rust mechanism that the borrow checker plays nice with.